builder/agent.py

```diff
@@ -45,11 +49,26 @@ def parse_args() -> tuple[_BuilderAgentArgs, list[str]]:
         parser.add_argument(
             '--chown', type=str,
             help='Reset owner of files to the specified `uid:gid`')
-
-        return parser.parse_known_args(namespace=_BuilderAgentArgs())
+        parser.add_argument(
+            '--env',
+            type=str,
+            action='append',
+            default=[],
+            help=(
+                'Environment variables to set (KEY=VALUE), '
+                'can be specified multiple times'
+            ),
+        )
+
+        return parser.parse_args(namespace=_BuilderAgentArgs())
 
     def main(self) -> None:
-        args, setup_args = self.parse_args()
+        args = self.parse_args()
+        env: dict[str, str] = {}
+
+        for kv in args.env:
+            k, v = kv.split('=')
+            env[k] = v
```

Comment:
Support cases where environment variable value contains `=`.

```suggestion
            pair = kv.split('=', 2)
            assert len(pair) == 2, 'invalid --env format'
            env[pair[0]] = pair[1]
```